h2o-core/src/main/java/water/parser/Parser.java

```diff
@@ -87,7 +87,10 @@ ParseWriter streamParseZip( final InputStream is, final StreamParseWriter dout,
     StreamParseWriter nextChunk = dout;
     int zidx = bvs.read(null,0,0); // Back-channel read of chunk index
     assert zidx==1;
-    while( is.available() > 0 ) {
+    int streamAvailable = is.available();
+    while(streamAvailable > 0) {
+      parseChunk(cidx++, din, nextChunk);
+      streamAvailable = is.available(); // Can (also!) rollover to the next input chunk
```

Comment:
is.available() can itself fetch another (source, compressed chunk). That is whole issue with the previous implementation. If that happens and we were processing the last chunk we will produce one less parsed chunk because the rollover to next chunk in is.available is never detected.